<a href="https://colab.research.google.com/github/KaiqueS14/projeto-salarios-dados/blob/main/Salarios.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### **Carregar as bibliotecas**

In [ ]:
import pandas as pd

###**Carregar a tabela e analisar a estrutura dos dados**

In [ ]:
df = pd.read_csv('webscraping.csv')
df.head(10)

JobTitle  ...                       Salary
0  novaEstágio em Business Intelligence | Foco em...  ...                Não informado
1               novaEstágio em Business Intelligence  ...                Não informado
2              novaAnalista de Business Intelligence  ...  R$ 2.585 - R$ 7.233 por mês
3           Estagiário de Business Intelligence (BI)  ...                Não informado
4           Analista de Business Intelligence Júnior  ...                Não informado
5                   Estágio em Business Intelligence  ...                Não informado
6                  Analista de Business Intelligence  ...  R$ 3.500 - R$ 5.000 por mês
7        Analista de Business Intelligence - Maracaí  ...                Não informado
8                  Analista de Business Intelligence  ...  R$ 3.000 - R$ 4.000 por mês
9                Estagiário de Business Intelligence  ...                Não informado

[10 rows x 7 columns]

####Quantidade de linhas e colunas

In [ ]:
df.shape

(837, 7)

####Tipagem de cada coluna

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 837 entries, 0 to 836
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   JobTitle     837 non-null    object
 1   Company      837 non-null    object
 2   Location     837 non-null    object
 3   PostDate     837 non-null    object
 4   ExtractDate  837 non-null    object
 5   Summary      837 non-null    object
 6   Salary       837 non-null    object
dtypes: object(7)
memory usage: 45.9+ KB


### **Tratamento da Tabela**

#### Transformar todas as colunas para letras maiusculas, facilitando a identificação de linhas duplicadas


In [ ]:
for coluna in df:
  df[coluna] = df[coluna].str.upper()

#### Identificando a existência de linhas duplicadas

In [ ]:
df.loc[df.duplicated()]

JobTitle  ...                           Salary
15                    GERENTE DE BUSINESS INTELLIGENCE  ...                    NÃO INFORMADO
60   BANCO ORIGINAL - ESPECIALISTA EM BUSINESS INTE...  ...                    NÃO INFORMADO
61                                NOVABUSINESS ANALYST  ...                    NÃO INFORMADO
105           NOVAESTÁGIO EM PESQUISA DE PÓS-GRADUAÇÃO  ...                    NÃO INFORMADO
106                     ARTIFICIAL INTELLIGENCE INTERN  ...                    NÃO INFORMADO
..                                                 ...  ...                              ...
814       LATAM ANALYTICS & OPERATIONS MANAGER, DRIVER  ...                    NÃO INFORMADO
829             GERENTE DE AUDITORIA CORPORATIVA DE TI  ...  R$ 200.000 - R$ 250.000 POR ANO
830           ANALISTA DE BI SAP SAC PLANNING - SÊNIOR  ...                    NÃO INFORMADO
831                        SECURITY ARCHITECT - SENIOR  ...                    NÃO INFORMADO
832             GERENTE DE AUDITORIA CORPORATIVA DE TI  ...  R$ 200.000 - R$ 250.000 POR ANO

[74 rows x 7 columns]

In [ ]:
qtd_duplicidades = df.loc[df.duplicated()].shape[0]
qtd_total = df.shape[0]
percentual_duplicidades = round(qtd_duplicidades/qtd_total,2)

print('Identificado um total de', qtd_duplicidades, 'linhas duplicadas. Isso representa', 
      100-percentual_duplicidades,'% de aproveitamento do dataset')

Identificado um total de 74 linhas duplicadas. Isso representa 99.91 % de aproveitamento do dataset


#### Criar novo df sem as duplicidades

In [ ]:
df_1 = df.drop_duplicates()
df_1.shape

(763, 7)

In [ ]:
df_1.head()

JobTitle  ...                       Salary
0  NOVAESTÁGIO EM BUSINESS INTELLIGENCE | FOCO EM...  ...                NÃO INFORMADO
1               NOVAESTÁGIO EM BUSINESS INTELLIGENCE  ...                NÃO INFORMADO
2              NOVAANALISTA DE BUSINESS INTELLIGENCE  ...  R$ 2.585 - R$ 7.233 POR MÊS
3           ESTAGIÁRIO DE BUSINESS INTELLIGENCE (BI)  ...                NÃO INFORMADO
4           ANALISTA DE BUSINESS INTELLIGENCE JÚNIOR  ...                NÃO INFORMADO

[5 rows x 7 columns]

#### Verificando a quantidade de vagas que possuem o mesmo nome/cargo

In [ ]:
df_1['JobTitle'].value_counts(dropna = False).reset_index()

index  JobTitle
0                                       ANALISTA DE BI        21
1                    ANALISTA DE BUSINESS INTELLIGENCE         8
2                                   ESPECIALISTA DE BI         8
3                                    COORDENADOR DE BI         6
4                                    ANALISTA DE BI SR         5
..                                                 ...       ...
669                            NOVAANALISTA BI - PLENO         1
670  ESTAGIÁRIO DE BUSINESS INTELLIGENCE & ALOCAÇÃO...         1
671  BANCO ORIGINAL - ESPECIALISTA EM BUSINESS INTE...         1
672         CONSULTANT (TECHNICAL) - JAPANESE SPEAKING         1
673  ESTAGIÁRIO DE WEB ANALYTICS / BUSINESS INTELLI...         1

[674 rows x 2 columns]

##### **De 674 vagas/cargos, a maioria possui nome unico. Portanto, é viável o agrupamento dos títulos para melhor análise**


#### Listando os 50 primeiros titulos de trabalho por ordem alfabética para identificar oportunidade de agrupamento

In [ ]:
pd.Series(df_1['JobTitle'].unique()).sort_values().head(50)

374                     2022 INTERNSHIP PROGRAM - BRAZIL
139                            3301 - CIENTISTA DE DADOS
140                                  3382 - CONSULTOR BI
256                                   4022 - ANALISTA BI
316                                 716 - ANALISTA DE BI
334                   ACCOUNT DEVELOPMENT REPRESENTATIVE
375                                  ACCOUNT EXECUTIVE 6
289             ACCOUNT EXECUTIVE CHANNELS - SISW 269647
236                    ADMINISTRATIVE ASSISTANT - SENIOR
145                                    AGRONOMIC ANALYST
498                        AI & IOT DESIGN RESEARCH LEAD
295                                     AI/ML SPECIALIST
322    ALIBABA GROUP CATEGORY SPECIALIST/SENIOR SPECI...
651    ALIBABA GROUP CATEGORY SPECIALIST/SENIOR SPECI...
367    ALIBABA GROUP LOGISTICS & SELLER OPERATIONS SE...
134     ALIBABA GROUP SUPPLY CHAIN & LOGISTICS BD BRAZIL
654    ALIEXPRESS CATEGORY SPECIALIST/SENIOR SPECIALI...
365    ALIEXPRESS CATEGORY SPEC